pkg/cloudprovider/providers/azure/azure_standard.go

```diff
@@ -70,7 +70,7 @@ func (az *Cloud) getStandardMachineID(resourceGroup, machineName string) string
 	return fmt.Sprintf(
 		machineIDTemplate,
 		az.SubscriptionID,
-		resourceGroup,
+		strings.ToLower(resourceGroup),
```

Comment:
what about the below function, are you going to convert to lower case?
```
func (az *Cloud) getAvailabilitySetID(resourceGroup, availabilitySetName string) string {
	return fmt.Sprintf(
		availabilitySetIDTemplate,
		az.SubscriptionID,
		resourceGroup,
		availabilitySetName)
}
```

Reply:
Conversion is not needed for other IDs, only providerID should be handled.